### ISSUE TITLE
[REVIEW]: MIDI.jl: Intuitive handling of MIDI data made for musicians

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1166]
**Submitting author:** @Datseris (<a href="http://orcid.org/0000-0002-6427-2385">George Datserus</a>)
**Repository:** <a href="https://github.com/JuliaMusic/MIDI.jl" target ="_blank">https://github.com/JuliaMusic/MIDI.jl</a>
**Version:** v1.1.3
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** @ssfrr, @jfsantos
**Archive:** <a href="https://doi.org/10.5281/zenodo.2591437" target="_blank">10.5281/zenodo.2591437</a>

## Status

[![status](http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15/status.svg)](http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15"><img src="http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15/status.svg)](http://joss.theoj.org/papers/e0cfc67982f857ed96d906ff2266aa15)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JuliaMusic/MIDI.jl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ssfrr & @jfsantos, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @Kevin-Mattheus-Moerman know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaMusic/MIDI.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.1.3
- [x] **Authorship:** Has the submitting author (@Datseris) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JuliaMusic/MIDI.jl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v1.1.3
- [x] **Authorship:** Has the submitting author (@Datseris) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ssfrr, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris, @ssfrr, @jfsantos, this is where the review takes place. There are instructions at the top of this issue as well as a check list which will guide you through the process. If all boxes are ticked and reviewers recommend acceptance this submission will be accepted in JOSS. 

You can comment on minor things here and we encourage your to use check lists (`- [ ]` in markdown). Larger/longer issues are better discussed in a separate issue which you can open on the [submission repository](https://github.com/JuliaMusic/MIDI.jl) and refer to here. 

Thanks again for helping with the review process!!!!!

Let me know if you have any questions.

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris can you work on fixing that figure in the paper? Let me know if you need help.

**Comment by Datseris** [comment]
@Kevin-Mattheus-Moerman I do not know how to fix this figure, as I said in a previous comment in the previous issue. It is a standard `.png` figure that is included with the standard `![description](figure)` format. The figure doesn't have overly big whitespace either.

From my perspective this is a problem of the "compilation/pdf-generation" procedure. If you know what I have to do then please let me know.

**Comment by Kevin-Mattheus-Moerman** [comment]
@arfon any idea what is causing the figure to be off center in [this paper](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)?

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris looks a bit like it is an inline picture so I tried to add extra lines in the PR https://github.com/JuliaMusic/MIDI.jl/pull/111.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Thank you Kevin, it worked. I'll add a better caption now...

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by jfsantos** [comment]
I will probably get to it around the middle of next week.

--
João Felipe Santos

**Comment by Datseris** [comment]
Hello @Kevin-Mattheus-Moerman , @jfsantos , @ssfrr . While I was trying to improve MIDI.jl I realized that I had to do a breaking bugfix. If you are interested you may look at https://github.com/JuliaMusic/MIDI.jl/pull/112 but this is in no way relevant to the paper.

Although this is a breaking change it is a bugfix and thus does not require a major version increment. Regardless, I have released version 1.1 that includes a warning of this change. if possible, please associate this paper with version 1.1 and not with version 1.0. Thank you.

**Comment by Kevin-Mattheus-Moerman** [comment]
@ssfrr, @jfsantos can you give an update on the review process? Thanks!

**Comment by ssfrr** [comment]
Hi Kevin, sorry for the delay. I should be able to take a look at this next week

**Comment by jfsantos** [comment]
I will work on it next Monday. Sorry for the delay.

--
João Felipe Santos

**Comment by Datseris** [comment]
Uhm, excuse me, but I believe I have to bump this.

**Comment by Kevin-Mattheus-Moerman** [comment]
@ssfrr, @jfsantos sorry to bother you again but it would be great if you could get started with this review. Thanks!

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris apologies for the delay encountered so far

**Comment by Datseris** [comment]
@Kevin-Mattheus-Moerman it's no problem, and I certainly hope I didn't come as across as impolite! It's because (I assume) we are all doing these things voluntarily as side projects that it is easy to forget things. So a reminder can be helpful in such cases.

**Comment by jfsantos** [comment]
Hello all, I just started my review. The version I get from Github is 1.1.0 but the review form says 1.0.0. The paper doesn't mention a specific version, so I guess that is a non-issue?

**Comment by jfsantos** [comment]
Other minor issues: 

- [ ] Most of the references are missing a DOI.
- [ ] Figure 1 is mentioned in the text as appearing right below the text, but is actually rendered on the next page. Please fix the text and point the reader to Figure 1 instead of assuming it is going to be rendered on a specific position.

**Comment by jfsantos** [comment]
Another question:

- The paper only describe examples of note events on MIDI, but the protocol/format deals with a lot more than just note events (like velocity changes, communication with multiple instruments in different channels, etc). Does the software address these other events? Is it future work or do the maintainers have no interest in supporting those?

**Comment by Datseris** [comment]
Thank you for the review @jfsantos .

> Figure 1 is mentioned in the text as appearing right below the text, but is actually rendered on the next page. Please fix the text and point the reader to Figure 1 instead of assuming it is going to be rendered on a specific position

Please read the latest version of the pdf, where this issue is fixed. whendon will compile a new one for you now. Towards my side there is only two issues: remove references with missing DOIs and answer the question:

> The paper only describe examples of note events on MIDI, but the protocol/format deals with a lot more than just note events (like velocity changes, communication with multiple instruments in different channels, etc). Does the software address these other events? Is it future work or do the maintainers have no interest in supporting those?

The core MIDI protocol is already implemented in MIDI.jl. These details far exceed the scope of a publication and are not discussed in the paper (and I believe I am already at the limit of word count suggested by JOSS).

Please read the documentation for more: in short, all MIDI/META/SysEx events are possible to write and read using MIDI.jl as well as any instrument channel (which as a result covers the entire MIDI protocol).

Anything exceeding what is already implemented is not of interest for the developer of MIDI.jl.

**Comment by Datseris** [comment]
@whendon generate pdf

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by jfsantos** [comment]
> Please read the documentation for more: in short, all MIDI/META/SysEx events are possible to write and read using MIDI.jl as well as any instrument channel (which as a result covers the entire MIDI protocol).

Maybe just mention this somewhere in the paper or documentation? I tried to find it but on a quick search did not find anything mentioning events that were not notes.

> Anything exceeding what is already implemented is not of interest for the developers of MIDI.jl.

Sounds good, and I don't think this needs to be mentioned anywhere.

**Comment by Datseris** [comment]
> Maybe just mention this somewhere in the paper or documentation? I tried to find it but on a quick search did not find anything mentioning events that were not notes.

I am not sure I agree with this statement, because this is already mentioned in the documentation. Here is my justification: 

Searching the documentation (using the search bar) for "META EVENT" brings me here: https://juliamusic.github.io/JuliaMusic_documentation.jl/dev/midi/io/#MIDI.TrackEvent which lists clearly (at least I believe so) the existence of and how to create or read META/MIDI/SysEx events. 

If one can create any of these three track events then one can implement the entire midi protocol.

---

Please let me know if this is still unclear.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Alright, I have fixed the missing DOIs. Now all cited literature includes a DOI.

**Comment by jfsantos** [comment]
I updated the checkboxes for the minor issues and review accordingly. I don't know what are the specific guidelines for versioning on JOSS but the paper does not describe which version of MIDI.jl it is talking about.

**Comment by Datseris** [comment]
Ideally I would like this paper to be associated with version 1.1.0. I have added an explicit statement about this in the manuscript now. Of course, it falls into the hands of @Kevin-Mattheus-Moerman whether this will be accepted and further more implemented, as it falls beyond my power to make such a change on `whedon`.

**Comment by ssfrr** [comment]
Getting rolling now, basing my review on the proof from [3 days ago](https://github.com/openjournals/joss-reviews/issues/1166#issuecomment-464238975)

**Comment by ssfrr** [comment]
This paper also references and uses MusicManipulations.jl. Is that in-scope for the review, or are we solely considering MIDI.jl?

**Comment by ssfrr** [comment]
## Review Summary:

MIDI.jl is a Julia package to enable users to read and write MIDI notes and tracks. While there is prior work in this space, it is useful to have a pure-julia implementation for working with MIDI files, as MIDI is a widely-used format for symbolic music representation. This work also includes the MusicManipulations package, which provides additional functionality for quantizing MIDI notes to a predefined grid, translating and transposing arrays of notes, filtering notes, and extracting individual properties into separate time series.

## Specific Issues

-   [ ] The paper makes the claim "This makes reading MIDI data easier than ever before.", which seems a little promotional and isn't substantiated.
-   [ ] There are multiple references to saving the user from worrying about understanding the byte-level representation of MIDI files. I think this can be assumed of any package that parses formatted data files and can be removed.
-   [ ] Most of the references are to papers on microtiming, which doesn't seem closely tied to MIDI.jl (though possibly to MusicManipulations.jl). I think these can be pared down to only what's relevant to the given example.
-   [ ] There's unreferenced prior art, e.g. [mido](https://github.com/mido/mido), [python-midi](https://github.com/vishnubob/python-midi), [pretty-midi](https://github.com/craffel/pretty-midi) and the [accompanying paper](https://colinraffel.com/publications/ismir2014intuitive.pdf). That paper also has some references to other packages (as of 2014). It would be useful to see some more discussion of how MIDI.jl compares in approach to these prior libraries. There's also [this MATLAB implementation](http://kenschutte.com/midi), but given the dependence on MATLAB and the relative complexity of calling into MATLAB relative to Python that is a less compelling comparison. The microtiming example could be slimmed down to compensate.

**Comment by Datseris** [comment]
Thank you @ssfrrr for your review. Let me know answer one by one your specific issues:

1. Thank you for pointing this out, I have now reworked my introduction.
2. Please be more specific about this issue. By reading the entire paper I can only find a singular instance: the opening paragraph of the section "Intuitive and simple interface". I could not find any other instance where there are "references to saving the user from worrying about understanding the byte-level representation of MIDI files.". If you believe there are multiple instances of this event please do state them explicitly. In addition, this software (and thus the paper) does not target audio engineers but musicians and scientists. Therefore it cannot be assumed that our users will "just know that they don't have to know about low level representations". We believe this has to be explicitly stated once, and we do so in our paper, again only once.
3. The editor of this review, @Kevin-Mattheus-Moerman asked me explicitly to include a "clear research application" in our paper, see the comment here: https://github.com/openjournals/joss-reviews/issues/1149#issuecomment-452094559 . As a scientist actively doing research in this field and actively publishing papers in this field I strongly stand that the current number of references should not be reduced further as they are all fully relevant to the example scientific application.
4. Thank you very much for sharing these references. I have included all of them in the paper (the MATLAB implementation was disregarded) and I have also added notes that compare **MIDI.jl** to them, in a concluding paragraph.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by ssfrr** [comment]
I updated the checklist of issues in [my comment above](https://github.com/openjournals/joss-reviews/issues/1166#issuecomment-464843735).

Please cite their paper as a real reference

Thank you for adding the comparison. Can you clarify items 1 and 2 in the "notable differences" list? Specifically:

> The Notes data structure and getnotes functionality that exists in MIDI.jl

Can you expand on how the data structure and note retrieval functionality are different in MIDI.jl from the other libraries? It seems that the main advantage over the other libraries is that MIDI.jl combines the note-on and note-off events into a single note object with a duration (which is definitely a useful feature of MIDI.jl), but pretty-midi seems to have a [similar representation](http://craffel.github.io/pretty-midi/#pretty-midi-note). If that's the main difference maybe clarify that here.

> The extensibility of MIDI.jl into higher level applications not limited only to raw midi data (see MusicManipulations.jl).

Here again pretty-midi seems to provide some high-level functionality, for instance tempo and beat estimation and note transition matrices.

**Comment by Datseris** [comment]
Thank you for your reply. 

> Please cite their paper as a real reference

Done.

> Can you expand on how the data structure and note retrieval functionality are different in MIDI.jl from the other libraries?

For **MIDI.jl** the `Note` is a central structure used everywhere and expanded by other modules. It has a much more important place than just a utility structure. Many functions (as shown in our official documentation) dispatch directly on the `Note` structure. In addition it has the `channel` property which I could not find elsewhere. Other than that indeed the "pretty-midi" representation seems quite similar. The paper was updated accordingly.

> Here again pretty-midi seems to provide some high-level functionality, for instance tempo and beat estimation and note transition matrices.

I have changed the sentence to:

> 2. **MIDI.jl** is extended further into higher level applications like the ones offered by **MusicManipulations.jl** or the module `MotifSequenceGenerator` that can create specially random sequences of notes.

I believe that pretty-midi is not extended by neither **MusicManipulations.jl** nor `MotifSequenceGenerator` and thus this is a notable difference.

---

@Kevin-Mattheus-Moerman can you please resolve the conflict regarding the software version?

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by ssfrr** [comment]
The reference you have in your references section is for the wrong paper. Also the link you gave is incorrect, and gives a 404. They provide the correct citation as well as a link to the paper in their [README](https://github.com/craffel/pretty-midi).

Please be more careful with your references. As I'm sure you're aware, properly citing prior work is one of the more important aspects of academic publishing. It shouldn't take three rounds of review to get directly-relevant prior work cited.

**Comment by Datseris** [comment]
> Please be more careful with your references. As I'm sure you're aware, properly citing prior work is one of the more important aspects of academic publishing. It shouldn't take three rounds of review to get directly-relevant prior work cited.

Wow, sorry for the mistake I did not realize this.

Unfortunately they do not provide the correct citation in their README as you claim. They provide a PDF file and a claim of publication. Of course I tried to find a DOI or an ISBN if the citation is for a book chapter. I have visited the official page for the conference the authors claim to have published in. It is here: https://dblp2.uni-trier.de/db/conf/ismir/ismir2014.html

As you will see for yourself, there is no paper titled "Intuitive Analysis, Creation and Manipulation of MIDI Data with pretty_midi" (but there is a single paper by C. Raffel which I wrongly assumed the correct one). In addition, searching the entire database for author "Colin Raffel" again does not return the claimed paper. You can see this here: https://dblp2.uni-trier.de/pers/hd/r/Raffel:Colin

If you do know where this paper is published, please be kind and share it with us here. The "text" they provide in the README is not enough of a proof to create a proper BiBTeX entry as required per the JOSS standard.

**Comment by ssfrr** [comment]
It appears that the short paper was accepted in the Late-Breaking/Demo category. The official conference website has a list of these accepted papers:

http://www.terasoft.com.tw/conf/ismir2014/AcceptedLBDPapers.html

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Thank you @ssfrr for helping me with this. I've now corrected the citation (which as a result also corrected the wrong link).

I have to note that even though I searched the website, I couldn't find an associated DOI for this paper (I don't think one exists). But I hope this is okay because there is at list a link that links to the paper.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by ssfrr** [comment]
The title for the raffel citation isn't showing up in the references. Maybe something weird with the bibtex?

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Alright, no matter what I try the title of the paper does not appear in the compiled pdf. If anyone knows the solution please let me know as well. The Bib entry I have now (which I stripped to the bare basics) is:
```
@article{Raffel,
  author    = {Raffel, Colin and Ellis, Daniel P. W.},
  title     = {{Intuitive Analysis, Creation and Manipulation of MIDI Data with pretty_midi}},
  journal   = {Proceedings of the 15th International Society for Music Information
               Retrieval Conference Late Breaking and Demo Papers, {ISMIR} 2014, Taipei, Taiwan, October 27-31,
               2014},
  year      = {2014},
  url       = {http://www.terasoft.com.tw/conf/ismir2014/LBD/LBD29.pdf},
}
```

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Fiiiiiiiiiiiiiinaly. It is not possible to have underscore in names in latex apparently.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@ssfrr, @jfsantos thanks for your review work!! Please can you check if you can tick those final boxes? 

Note that the version will be updated before acceptance so that box (which has been removed in future review issues) can safely be ticked. 

@ssfrr, can you check the reference which should now be in good shape?

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check DOIs

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by Kevin-Mattheus-Moerman** [comment]
Some minor editorial comments on the paper

- [ ] Can  you check the sentence "A  music note can be (in its most basic level) deconstructed into just four numbers: the temporal position that the note is played, the duration, the pitch and the intensity (strength with which the note is played, also called velocity)". Perhaps it should read: "A music note can be (in its most basic level) deconstructed into just four numbers: the temporal position **~~that~~** the note is played **in**, the duration, the pitch[,] and the intensity (strength with which the note is played, also called velocity)"

- [ ] "This easy to use high level interface allows MIDI.jl to be extendable. In e.g. another
software package MusicManipulations.jl we provide general functions for manipulating
(and further analyzing) music data.
For example, the function quantize from the package MusicManipulations.jl allows
the user to quantize any Notes instance to any grid." I propose to remove "this", to rephrase the part with "e.g.", and to remove the enter/paragraph break. i.e.: -> "**The** easy to use high level interface allows MIDI.jl to be extendable. For instance ~~ In e.g~~, in another software package MusicManipulations.jl we provide general functions for manipulating (and further analyzing) music data. For example, the function quantize from the package MusicManipulations.jl allows the user to quantize any Notes instance to any grid."

- [ ] Check the use of "high level" and "high-level" please use the hyphen consistently

- [ ] Add comma after "In addition" in "In addition[,] it has plenty more use for scientific applications."

- [ ] Please remove: ", which also has an associated publication" (i.e. the citation alone is sufficient)

- [ ] Consider rephrasing the following: "Something similar exists in pretty-midi as well but for example there is no channel property." e.g. to something like "Although the package pretty-midi contains similar functionality, it lacks the channel property."

- [ ] If DAW is an acronym can you please define it? 

- [ ] Consider rephrasing "There is also no sequencer functionality in MIDI.jl, which is in example in python-midi." to something along the lines of: "Sequencer functionality, which has been implemented in for instance the python-midi package, is currently lacking in MIDI.jl"

- [ ] The following sentence sounds a bit vague, can you rephrase or perhaps expand with more examples too: "pretty-midi has some other high level functionality (like e.g. basic tempo estimation) also absent in MIDI.jl.", perhaps this improves the sentence especially if you expand it a bit: "MIDI.jl currently lacks some basic high-level functionality such as tempo estimation, which have been implemented in packages like pretty-midi."

- [ ] For the reference "Does it swing? Microtiming deviations and swing feeling in jazz", I would recommend (if possible) to upload it as a pre-print on some pre-print server (if allowed by publisher see here: http://www.sherpa.ac.uk/romeo/index.php), and to cite the DOI of the pre-print in this paper.

**Comment by Datseris** [comment]
@Kevin-Mattheus-Moerman thank you for your reply.

I have implemented all your suggestions besides the final one. Unfortunately the research group associated with this publication has decided to _not_ publish it in a non peer-reviewed outlet (such as pre-print servers). In addition the paper is currently under review in a peer-reviewed journal, which makes me believe there is no reason to waste resources into putting it on some archive.

But there is no problem. If JOSS does not allow citing to-be-published work I can simply remove the paragraph that references this work. I believe the rest of the manuscript already makes a strong case for the scientific relevance of **MIDI.jl**.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
> @Kevin-Mattheus-Moerman thank you for your reply.
> 
> I have implemented all your suggestions besides the final one. Unfortunately the research group associated with this publication has decided to _not_ publish it in a non peer-reviewed outlet (such as pre-print servers). In addition the paper is currently under review in a peer-reviewed journal, which makes me believe there is no reason to waste resources into putting it on some archive.
> 
> But there is no problem. If JOSS does not allow citing to-be-published work I can simply remove the paragraph that references this work. I believe the rest of the manuscript already makes a strong case for the scientific relevance of **MIDI.jl**.

That is fine. The pre-print is only a recommendation so you can leave it as is. Thanks.

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris I've gone through your revisions. There is one more point remaining: 

- [ ] Please remove: ", which also has an associated publication" (i.e. the citation alone is sufficient)

**Comment by Datseris** [comment]
> That is fine. The pre-print is only a recommendation so you can leave it as is. Thanks.

Thank you very much for understanding.

> Please remove: ", which also has an associated publication" (i.e. the citation alone is sufficient)

Ops, so sorry I forgot this. It is fixed now!

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by ssfrr** [comment]
LGTM. boxes checked.

**Comment by Kevin-Mattheus-Moerman** [comment]
Thanks @ssfrr

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- http://doi.org/10.1137/141000671 is OK
- http://doi.org/10.1037/a0024323 is OK
- http://doi.org/10.1525/mp.2010.27.3.157 is OK
- http://doi.org/10.1177/1029864913486793 is OK
- http://doi.org/10.1525/mp.2013.30.5.497 is OK
- http://doi.org/10.3389/fpsyg.2016.01487 is OK
- http://doi.org/10.1080/09298215.2017.1355394 is OK
- http://doi.org/10.1177/0305735617697507 is OK
- http://doi.org/10.1177/0305735617697507 is OK
- http://doi.org/10.1371/journal.pone.0026457 is OK
- http://doi.org/10.1371/journal.pone.0127902 is OK
- http://doi.org/10.1371/journal.pone.0186361 is OK
- http://doi.org/10.1371/journal.pone.0127902 is OK
- http://doi.org/10.1073/pnas.1324142111 is OK

MISSING DOIs

- https://doi.org/10.1371/journal.pone.0186361 may be missing for title: Does it Swing? Microtiming Deviations and Swing Feeling in Jazz
- https://doi.org/10.2307/852199 may be missing for title: Searching for Swing: Participatory Discrepancies in the Jazz Rhythm Section
- https://doi.org/10.1037/a0024208 may be missing for title: Sensorimotor coupling in music and the psychology of the groove.
- https://doi.org/10.1525/mp.2016.34.2.123 may be missing for title: The Asymmetrical Influence of Timing Asynchrony of Bass Guitar and Drum Sounds on Groove
- https://doi.org/10.3389/fpsyg.2015.01232 may be missing for title: Microtiming in Swing and Funk affects the body movement behavior of music expert listeners
- https://doi.org/10.1093/gmo/9781561592630.article.j333500 may be missing for title: Gridless Beats
- https://doi.org/10.1525/mts.2011.33.1.3 may be missing for title: Why do jazz musicians swing their eighth notes?
- https://doi.org/10.1163/22134468-00002043 may be missing for title: Moving to the Beat: Studying Entrainment to Micro-Rhythmic Changes in Pulse by Motion Capture
- https://doi.org/10.3389/fpsyg.2017.01709 may be missing for title: Rhythmic Density Affects Listeners’ Emotional Response to Microtiming
- https://doi.org/10.1525/mp.2002.19.3.387 may be missing for title: Embodied mind, situated cognition, and expressive microtiming in African-American music
- https://doi.org/10.1525/can.1987.2.3.02a00010 may be missing for title: Participatory discrepancies and the power of music
- https://doi.org/10.2307/427969 may be missing for title: Motion and Feeling through Music
- https://doi.org/10.1525/mp.2002.19.3.333 may be missing for title: Swing ratios and ensemble timing in jazz performance: Evidence for a common rhythmic pattern
- https://doi.org/10.2466/pms.1991.73.3.707 may be missing for title: An analysis of “swing” subdivision and asynchronization in three jazz saxophonists
- https://doi.org/10.1177/0305735614555790 may be missing for title: Timing deviations in jazz performance: The relationships of selected musical variables on horizontal and vertical timing relations: A case study
- https://doi.org/10.1525/mts.2009.31.1.1 may be missing for title: Time warps in early jazz
- https://doi.org/10.1016/j.cortex.2008.06.011 may be missing for title: On the role and origin of isochrony in human rhythmic entrainment
- https://doi.org/10.1525/mp.2007.25.1.43 may be missing for title: Perception and production of syncopated rhythms
- https://doi.org/10.1162/jocn.2007.19.5.893 may be missing for title: Rhythm and beat perception in motor areas of the brain
- https://doi.org/10.1093/gmo/9781561592630.article.27263 may be missing for title: Syncopation and the Score
- https://doi.org/10.2478/v10053-008-0094-0 may be missing for title: Cognitive and affective judgements of syncopated musical themes
- https://doi.org/10.3389/fpsyg.2014.01036 may be missing for title: Syncopation creates the sensation of groove in synthesized music examples
- https://doi.org/10.1371/journal.pone.0094446 may be missing for title: Syncopation, Body-Movement and Pleasure in Groove Music
- https://doi.org/10.1525/mp.2009.26.5.401 may be missing for title: Fractal tempo fluctuation and pulse prediction
- https://doi.org/10.1080/01621459.1977.10479969 may be missing for title: Wald’s test as applied to hypotheses in logit analysis
- https://doi.org/10.2307/2531595 may be missing for title: Comparing the areas under two or more correlated receiver operating characteristic curves: a nonparametric approach
- https://doi.org/10.1525/mp.2007.25.1.43 may be missing for title: Perception and production of syncopated rhythms
- https://doi.org/10.1177/1029864914541851 may be missing for title: Der Gold-MSI: replikation und validierung eines fragebogeninstrumentes zur messung musikalischer erfahrenheit anhand einer deutschen stichprobe
- https://doi.org/10.1371/journal.pone.0089642 may be missing for title: The musicality of non-musicians: an index for assessing musical sophistication in the general population

INVALID DOIs

- None
```

**Comment by Datseris** [comment]
@Kevin-Mattheus-Moerman am I supposed to do something? From the compiled pdf I can see that all articles that do have a DOI display it properly. I think whedon is pointing out references that I have in the `.bib` file but I don't actively use in the article.

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris okay. Thanks for clarifying that. If you clean up the .bib file we can avoid this error but it sounds like your paper is in good shape.

**Comment by Kevin-Mattheus-Moerman** [comment]
@Datseris looks like we are nearly there. I will soon recommend that this work is accepted in JOSS. At this point, can you please:

- [ ] Archive a copy of the reviewed software on a service like [Zenodo](https://zenodo.org/) (some follow [these steps](https://guides.github.com/activities/citable-code/)) and list the DOI of this archived version here?

- [ ] Provide the version/release tag for the reviewed and archived version? E.g. it likely moved on from v1.0.0

Thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@ssfrr, @jfsantos thank you for your review work!!! :tada:

**Comment by Datseris** [comment]
Thanks for the support @Kevin-Mattheus-Moerman , here is the DOI from ZENODO:



[![DOI](https://zenodo.org/badge/41460547.svg)](https://zenodo.org/badge/latestdoi/41460547)



The version is `v1.1.3`

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo.2591437 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.2591437" target="_blank">10.5281/zenodo.2591437</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set v1.1.3 as version

**Comment by whedon** [comment]
OK. v1.1.3 is the version.

**Comment by Kevin-Mattheus-Moerman** [comment]
The DOI link produces an error but it does seem correct: https://zenodo.org/record/2591437#.XIfeO4WnxBI. Perhaps it takes a while to register properly as you've just created it.

**Comment by Kevin-Mattheus-Moerman** [comment]
@openjournals/joss-eics this paper is ready to be accepted

**Comment by labarba** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by labarba** [comment]
Hi @Datseris — I'm going to ask you to edit the metadata of the Zenodo deposit (no need to get new version or new DOI) so the title and author list match the JOSS paper.

**Comment by labarba** [comment]
I do have some editorial suggestions on the paper:

- Documentation: "very well documented" >> remove "very" 
- "The documentation link can be found here." >> It's awkward to have "here" as an active link in a PDF (or even a website, where it's considered bad form!). You could add the URL to the references list, instead? Or simply spell out the full URL (or create a short URL) in the body text.
- "Because of the rich documentation, in tis paper…" >> this sentence is unnecessary, because all JOSS papers are requested _not_ to include any documentation.
- Given the above notes, I recommend removing the heading "Documentation" and merging the sentences left to the "Introduction" section.
- various places: high-level interface, low-level interface missing the hyphen in the compound adjective
- various missing commas, after "In this paper," after "In addition," after "e.g.," and "In the following."
- "the data structure we create" >> unclear: is it the data structure _you_ (the authors) created as part of your design process? (in which case it should be in past sense), or is ti the data structure that the software creates when it is being used? (in which case the subject of the sentence is not "we") 
- "The easy to use high level interface" >> missing hyphens in compound adjectives
- Scientific Application: "see (references)">> if you are citing references as parts of a sentence, please use the Author (Year) format:  For the syntax to obtain brackets in the right places, see: 
https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.htm
- Necessity of MIDI.jl — the fact that no other package offers the functionally does not make it necessary! I think you can remove the heading here, and move that lonely sentence into the Related Software section. (The case for need of the software was made in the previous section.)
- Do a search for the word "very" and in each appearance ask yourself if you really need it.

Finally, you are citing _"Does it swing? Microtiming deviations and swing feeling in jazz"_, which is apparently unpublished. Is it possible to deposit the preprint somewhere? (Like OSF preprints, or similar.) Citation to a non-public artifact is discouraged.

**Comment by Datseris** [comment]
Hi @labarba ,

In the latest master commit I implemented all your points.

> Scientific Application: "see (references)">> if you are citing references as parts of a sentence, please use the Author (Year) format: For the syntax to obtain brackets in the right places, see:
https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.htm

I don't understand where you are pointing at... In text I have:
 
```
These have been studied extensively in the literature and their importance and influence are debated strongly, see  [@Madison2011, @Butterfield2010, @Fruehauf2013, @Davies2013, @Senn2016, @Hofmann2017] and references therein.
```

every reference is named. In the pdf all references are expanded to the Author (Year) format.

---

> Finally, you are citing "Does it swing? Microtiming deviations and swing feeling in jazz", which is apparently unpublished. Is it possible to deposit the preprint somewhere? (Like OSF preprints, or similar.) Citation to a non-public artifact is discouraged. 

No, this is not possible, please see the discussion I had with @Kevin-Mattheus-Moerman  in the comment above:  https://github.com/openjournals/joss-reviews/issues/1166#issuecomment-468617973

> Citation to a non-public artifact is discouraged. 

That may be true, but our group discourages non peer reviewed publications. We would appreciate your respecting of our choice.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by labarba** [comment]
Regarding the citations, whether in-text or not, imagine if the format of the article used superscripts for citations: it would be quite weird to see a verb with a superscript but no subject, right? So, when you use the citation as part of speech in a sentence, you display it differently than if the reference is parenthetical (and could be replaced with a footnote and not look weird). 

See: https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.htm#citation_syntax
... the part about in-text citations.

**Comment by labarba** [comment]
Regarding your decision _not_ to post a pre-print of the other manuscript, that is of course your prerogative. But in that case, please remove it from the references list. You can still mention it in the text, with something like: "we currently have a manuscript under review titled "Blah" that used this software." 

I'm still a little surprised by your answer, though. Pre-prints are increasingly recognized as a valid and important tool for open science. Most journals now accept pre-printing and don't count it as prior publication. Have a look at: https://en.wikipedia.org/wiki/List_of_academic_journals_by_preprint_policy

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
Hi @labarba ,

In the latest version I've tried to answer your comments again. By reading the "in-text citations" part what I've taken is that I should replace brackets with parenthesis.

**Comment by labarba** [comment]
> Hi @Datseris — I'm going to ask you to edit the metadata of the Zenodo deposit (no need to get new version or new DOI) so the title and author list match the JOSS paper.

Looks like this is still pending?

**Comment by labarba** [comment]
The documentation says:

You can also write an in-text citation, as follows:

`@smith04 says blah.`

No brackets.

**Comment by Datseris** [comment]
> Looks like this is still pending?

Yeap, fixed now. I had to publish the changes: https://zenodo.org/record/2591437

> No brackets.

I've removed all brackets from current version. Let's see:

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/548

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/548, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1137/141000671 is OK
- 10.1037/a0024323 is OK
- 10.1525/mp.2010.27.3.157 is OK
- 10.1177/1029864913486793 is OK
- 10.1525/mp.2013.30.5.497 is OK
- 10.3389/fpsyg.2016.01487 is OK
- 10.1080/09298215.2017.1355394 is OK
- 10.1177/0305735617697507 is OK
- 10.1177/0305735617697507 is OK
- 10.1371/journal.pone.0026457 is OK
- 10.1371/journal.pone.0127902 is OK
- 10.1371/journal.pone.0186361 is OK
- 10.1007/978-1-4612-0919-5_38 is OK
- 10.1371/journal.pone.0127902 is OK
- 10.1073/pnas.1324142111 is OK

MISSING DOIs

- https://doi.org/10.1371/journal.pone.0186361 may be missing for title: Does it Swing? Microtiming Deviations and Swing Feeling in Jazz
- https://doi.org/10.2307/852199 may be missing for title: Searching for Swing: Participatory Discrepancies in the Jazz Rhythm Section
- https://doi.org/10.1037/a0024208 may be missing for title: Sensorimotor coupling in music and the psychology of the groove.
- https://doi.org/10.1108/09504120410543273 may be missing for title: The Harvard Dictionary of Music: Fourth Edition
- https://doi.org/10.1525/mp.2016.34.2.123 may be missing for title: The Asymmetrical Influence of Timing Asynchrony of Bass Guitar and Drum Sounds on Groove
- https://doi.org/10.3389/fpsyg.2015.01232 may be missing for title: Microtiming in Swing and Funk affects the body movement behavior of music expert listeners
- https://doi.org/10.1093/gmo/9781561592630.article.j333500 may be missing for title: Gridless Beats
- https://doi.org/10.1525/mts.2011.33.1.3 may be missing for title: Why do jazz musicians swing their eighth notes?
- https://doi.org/10.30535/mto.22.1.6 may be missing for title: Metrical ambiguity or microrhythmic flexibility? Analysing groove in “Nasty Girl” by Destiny’s Child
- https://doi.org/10.1163/22134468-00002043 may be missing for title: Moving to the Beat: Studying Entrainment to Micro-Rhythmic Changes in Pulse by Motion Capture
- https://doi.org/10.3389/fpsyg.2017.01709 may be missing for title: Rhythmic Density Affects Listeners’ Emotional Response to Microtiming
- https://doi.org/10.1371/journal.pone.0186361 may be missing for title: Groove microtiming deviations as phase shifts
- https://doi.org/10.1525/mp.2002.19.3.387 may be missing for title: Embodied mind, situated cognition, and expressive microtiming in African-American music
- https://doi.org/10.1525/can.1987.2.3.02a00010 may be missing for title: Participatory discrepancies and the power of music
- https://doi.org/10.2307/427969 may be missing for title: Motion and Feeling through Music
- https://doi.org/10.1525/mp.2002.19.3.333 may be missing for title: Swing ratios and ensemble timing in jazz performance: Evidence for a common rhythmic pattern
- https://doi.org/10.2466/pms.1991.73.3.707 may be missing for title: An analysis of “swing” subdivision and asynchronization in three jazz saxophonists
- https://doi.org/10.1177/0305735614555790 may be missing for title: Timing deviations in jazz performance: The relationships of selected musical variables on horizontal and vertical timing relations: A case study
- https://doi.org/10.1525/mts.2009.31.1.1 may be missing for title: Time warps in early jazz
- https://doi.org/10.1016/j.cortex.2008.06.011 may be missing for title: On the role and origin of isochrony in human rhythmic entrainment
- https://doi.org/10.1525/mp.2007.25.1.43 may be missing for title: Perception and production of syncopated rhythms
- https://doi.org/10.1162/jocn.2007.19.5.893 may be missing for title: Rhythm and beat perception in motor areas of the brain
- https://doi.org/10.1093/gmo/9781561592630.article.27263 may be missing for title: Syncopation and the Score
- https://doi.org/10.2478/v10053-008-0094-0 may be missing for title: Cognitive and affective judgements of syncopated musical themes
- https://doi.org/10.3389/fpsyg.2014.01036 may be missing for title: Syncopation creates the sensation of groove in synthesized music examples
- https://doi.org/10.1371/journal.pone.0094446 may be missing for title: Syncopation, Body-Movement and Pleasure in Groove Music
- https://doi.org/10.1525/mp.2009.26.5.401 may be missing for title: Fractal tempo fluctuation and pulse prediction
- https://doi.org/10.1007/0-387-22702-4_4 may be missing for title: ordinal—Regression Models for Ordinal Data
- https://doi.org/10.2172/1234602 may be missing for title: MuMIn, Multi-model inference. R package version 1.15. 6.; 2016
- https://doi.org/10.1080/01621459.1977.10479969 may be missing for title: Wald’s test as applied to hypotheses in logit analysis
- https://doi.org/10.2307/2531595 may be missing for title: Comparing the areas under two or more correlated receiver operating characteristic curves: a nonparametric approach
- https://doi.org/10.1525/mp.2007.25.1.43 may be missing for title: Perception and production of syncopated rhythms
- https://doi.org/10.1177/1029864914541851 may be missing for title: Der Gold-MSI: replikation und validierung eines fragebogeninstrumentes zur messung musikalischer erfahrenheit anhand einer deutschen stichprobe
- https://doi.org/10.1371/journal.pone.0089642 may be missing for title: The musicality of non-musicians: an index for assessing musical sophistication in the general population
- https://doi.org/10.1002/sce.3730300211 may be missing for title: Big Book of Swing
- https://doi.org/10.1109/icassp.2016.7471641 may be missing for title: Intuitive analysis, creation and manipulation of MIDI data with pretty_midi

INVALID DOIs

- None
```

**Comment by labarba** [comment]
It looks like you have a bunch of entries in the `paper.bib` file that are not cited in the paper. I think you will need to clean up the `.bib` file, because these are listed as citations in the `xml` that we send to Crossref. See:
https://github.com/openjournals/joss-papers/blob/6d4ea9c3a81a9603b6532c3eb9466cc9b93f3d42/joss.01166/10.21105.joss.01166.crossref.xml

@arfon — Can you confirm that we need the `paper.bib` file to contain _only_ the references cited in the text? I'm not sure about this.

**Comment by arfon** [comment]
> @arfon — Can you confirm that we need the paper.bib file to contain only the references cited in the text? I'm not sure about this.

Yes please.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1137/141000671 is OK
- 10.1037/a0024323 is OK
- 10.1525/mp.2010.27.3.157 is OK
- 10.1177/1029864913486793 is OK
- 10.1525/mp.2013.30.5.497 is OK
- 10.3389/fpsyg.2016.01487 is OK
- 10.1080/09298215.2017.1355394 is OK
- 10.1371/journal.pone.0026457 is OK
- 10.1371/journal.pone.0127902 is OK
- 10.1371/journal.pone.0186361 is OK
- 10.1073/pnas.1324142111 is OK

MISSING DOIs

- https://doi.org/10.1109/icassp.2016.7471641 may be missing for title: Intuitive analysis, creation and manipulation of MIDI data with pretty_midi

INVALID DOIs

- None
```

**Comment by Datseris** [comment]
Perfect, I removed all unused references! (The reference you see above without DOI does not have an associated DOI to it)

**Comment by labarba** [comment]
Do you want to add caps protection, `{MIDI}`, in the title, so it appears in upper case?

**Comment by Datseris** [comment]
@labarba Thanks for the suggestion. I did that now in the latest commit.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by Datseris** [comment]
@labarba Putting `{}` in the title in the markdown file didn't work so I removed it. I'd prefer to continue with the publication process instead of adding caps protection.

**Comment by Datseris** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01166/joss.01166/10.21105.joss.01166.pdf)

**Comment by labarba** [comment]
I meant on the title of the reference we were talking about: 
https://github.com/JuliaMusic/MIDI.jl/blob/610f3a26085446e74282d026e2d6359b01194225/paper/paper.bib#L235

**Comment by Datseris** [comment]
Okay done this now @labarba .

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1137/141000671 is OK
- 10.1037/a0024323 is OK
- 10.1525/mp.2010.27.3.157 is OK
- 10.1177/1029864913486793 is OK
- 10.1525/mp.2013.30.5.497 is OK
- 10.3389/fpsyg.2016.01487 is OK
- 10.1080/09298215.2017.1355394 is OK
- 10.1371/journal.pone.0026457 is OK
- 10.1371/journal.pone.0127902 is OK
- 10.1371/journal.pone.0186361 is OK
- 10.1073/pnas.1324142111 is OK

MISSING DOIs

- https://doi.org/10.1109/icassp.2016.7471641 may be missing for title: Intuitive analysis, creation and manipulation of MIDI data with pretty_midi

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/552

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/552, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/553
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01166](https://doi.org/10.21105/joss.01166)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @Datseris, your JOSS paper is now published!

Sincere thanks to the editor: @Kevin-Mattheus-Moerman, and reviewers: @ssfrr, @jfsantos 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01166/status.svg)](https://doi.org/10.21105/joss.01166)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01166">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01166/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01166/status.svg
   :target: https://doi.org/10.21105/joss.01166
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01166/status.svg)](https://doi.org/10.21105/joss.01166)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=

**Comment by Datseris** [comment]
Thank you very much.

**Comment by labarba** [comment]
https://twitter.com/LorenaABarba/status/1106175091551936512

**Comment by Datseris** [comment]
I just want to be on the safe side, since doi2bib.org cannot find the DOI of this publication.

Is the following BibTeX entry the correct one?

```latex
@article{Datseris2019,
  doi = {10.21105/joss.01166},
  url = {https://doi.org/10.21105/joss.01166},
  year  = {2019},
  month = {mar},
  publisher = {The Open Journal},
  volume = {4},
  number = {35},
  pages = {1166},
  author = {George Datseris and Joel Hobson},
  title = {{MIDI}.jl: Simple and intuitive handling of MIDI data.},
  journal = {The Journal of Open Source Software}
}
```

**Comment by labarba** [comment]
The DOI resolves:
 https://doi.org/10.21105/joss.01166

**Comment by arfon** [comment]
According to https://www.doi2bib.org/bib/10.21105%2Fjoss.01166 (this looks correct to me):

```
@article{Datseris2019,
  doi = {10.21105/joss.01166},
  url = {https://doi.org/10.21105/joss.01166},
  year  = {2019},
  month = {mar},
  publisher = {The Open Journal},
  volume = {4},
  number = {35},
  pages = {1166},
  author = {George Datseris and Joel Hobson},
  title = {{MIDI}.jl: Simple and intuitive handling of {MIDI} data.},
  journal = {Journal of Open Source Software}
}@article{Datseris2019,
  doi = {10.21105/joss.01166},
  url = {https://doi.org/10.21105/joss.01166},
  year  = {2019},
  month = {mar},
  publisher = {The Open Journal},
  volume = {4},
  number = {35},
  pages = {1166},
  author = {George Datseris and Joel Hobson},
  title = {{MIDI}.jl: Simple and intuitive handling of {MIDI} data.},
  journal = {Journal of Open Source Software}
}
```
